framework/src/main/java/org/checkerframework/framework/flow/CFAbstractStore.java

```diff
@@ -1186,6 +1286,24 @@ private S upperBound(S other, boolean shouldWiden) {
       }
     }
 
+    for (Map.Entry<LocalVariable, BooleanVarStore<V, S>> e : other.booleanVarStores.entrySet()) {
+      // If a local variable appears in just one store, then the other store implicitly contains
```

Comment:
Here and above, can you remark on whether this is done for correctness, or performance, or as a heuristic?